src/rws_interface.cpp

```diff
@@ -82,6 +82,38 @@ RWSInterface::StaticInfo RWSInterface::collectStaticInfo()
   return static_info;
 }
 
+std::vector<cfg::moc::Transmission> RWSInterface::getCFGTransmission()
+{
+  std::vector<cfg::moc::Transmission> result;
+
+  RWSClient::RWSResult rws_result = rws_client_.getConfigurationInstances("MOC", "TRANSMISSION");
+
+  std::vector<Poco::XML::Node*> instances;
+  instances = xmlFindNodes(rws_result.p_xml_document, XMLAttributes::CLASS_CFG_DT_INSTANCE_LI);
+
+  for (size_t i = 0; i < instances.size(); ++i)
+  {
+    std::vector<Poco::XML::Node*> attributes = xmlFindNodes(instances[i], XMLAttributes::CLASS_CFG_IA_T_LI);
+
+    cfg::moc::Transmission transmission;
+
+    transmission.name = xmlNodeGetAttributeValue(instances[i], Identifiers::TITLE);
+
+    for (size_t j = 0; j < attributes.size(); ++j)
+    {
+      Poco::XML::Node* attribute = attributes[j];
+      if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, "rotating_move"))
+      {
+        transmission.rotating_move = (xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE) == "true");
```

Comment:
I haven't looked up what `xmlNodeHasAttribute(..)` does exactly, but is `attribute` equal to something which encodes for revolute joints here?

Reply:
Example of a transmission list item (collected from a virtual controller):

![Capture2](https://user-images.githubusercontent.com/16589371/97868004-3670a900-1d0f-11eb-98e4-535582ce6836.PNG)

The `xmlNodeHasAttribute(attribute, Identifiers::TITLE, "rotating_move")` just checks if the XML node (i.e. `attribute`) has the XML attribute `title="rotating_move"`, and `rotating_move` represents an attribute for a RobotWare transmission configuration instance.

Side note, the [getNodeByPath(...)](https://pocoproject.org/docs/Poco.XML.Node.html#32546) function can probably replace several of the custom functions for finding and checking XML nodes.